lib/archethic/p2p/client/connection.ex

```diff
@@ -77,8 +77,18 @@ defmodule Archethic.P2P.Client.Connection do
       availability_timer: {nil, 0}
     }
 
-    actions = [{:next_event, :internal, :connect}]
-    {:ok, :disconnected, data, actions}
+    # First connection attempt is synchronous (to deal with the `start_link` followed by a `send_message`)
+    # We call ourselves the `state_enter` because `GenStateMachine` will not do it from init
+    # `handle_connect` might take a long time
+    case transport.handle_connect(ip, port) do
+      {:ok, socket} ->
+        {_, new_data} = handle_event(:enter, :disconnected, {:connected, socket}, data)
```

Comment:
Using `handle_event` is not the most elegant, to avoid duplicate code, you could wrap the code in the `enter, :disconnected, {:connected, socket}, data` event in a private function

Reply:
As discussed with Sam, we may use the `:postpone` atom in gen_statem to deal with this in an elegant manner.